library/src/main/java/org/apache/apex/malhar/lib/window/impl/KeyedWindowedJoinOperatorImpl.java

```diff
@@ -0,0 +1,82 @@
+package org.apache.apex.malhar.lib.window.impl;
+
+import java.util.Map;
+
+import org.apache.apex.malhar.lib.window.JoinAccumulation;
+import org.apache.apex.malhar.lib.window.TriggerOption;
+import org.apache.apex.malhar.lib.window.Tuple;
+import org.apache.apex.malhar.lib.window.Window;
+import org.apache.apex.malhar.lib.window.WindowedStorage;
+
+import com.datatorrent.lib.util.KeyValPair;
+
+
+/**
+ * Keyed Windowed Join Operator
+ */
+public class KeyedWindowedJoinOperatorImpl<KeyT, InputT1, InputT2, AccumT, OutputT>
+    extends AbstractWindowedJoinOperator<KeyValPair<KeyT, InputT1>, KeyValPair<KeyT, InputT2>, KeyValPair<KeyT, OutputT>, WindowedStorage.WindowedKeyedStorage<KeyT, AccumT>, WindowedStorage.WindowedKeyedStorage<KeyT, OutputT>, JoinAccumulation<InputT1, InputT2, ?, ?, ?, AccumT, OutputT>>
+{
+  // TODO: Add session window support.
+
+  @Override
+  public void accumulateTuple2(Tuple.WindowedTuple<KeyValPair<KeyT, InputT2>> tuple)
+  {
+    KeyValPair<KeyT, InputT2> kvData = tuple.getValue();
+    KeyT key = kvData.getKey();
+    for (Window window : tuple.getWindows()) {
+      // process each window
+      AccumT accum = dataStorage.get(window, key);
+      if (accum == null) {
+        accum = accumulation.defaultAccumulatedValue();
+      }
+      dataStorage.put(window, key, accumulation.accumulate2(accum, kvData.getValue()));
+    }
+  }
+
+  @Override
+  public void accumulateTuple(Tuple.WindowedTuple<KeyValPair<KeyT, InputT1>> tuple)
+  {
+    KeyValPair<KeyT, InputT1> kvData = tuple.getValue();
+    KeyT key = kvData.getKey();
+    for (Window window : tuple.getWindows()) {
+      // process each window
+      AccumT accum = dataStorage.get(window, key);
+      if (accum == null) {
+        accum = accumulation.defaultAccumulatedValue();
+      }
+      dataStorage.put(window, key, accumulation.accumulate(accum, kvData.getValue()));
+    }
+  }
+
```

Comment:
I tried to do this, but because when the accumulation wants to do `accumulate()`/`accumulate2()`, it has to know the specific type of the value of the key value pair wrapped in the tuple, so giving it a type `T` instant will cause a type error.